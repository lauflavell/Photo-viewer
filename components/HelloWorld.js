import React from "react";

export function HelloWorld({ name }) {    // declare and export new function called 'PhotoViewer'  
        if (name)  {
            return (
            <div>
                <p>Hello, {name} </p>
            </div>)
        }
        return (
            <div>          
                <p> Hello, world</p> 
            </div>
        )
}