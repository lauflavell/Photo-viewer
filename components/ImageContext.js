import React from "react";

export const ImageContext = React.createContext({
    imgUrl: "https://picsum.photos/id/600/1600/900.jpg",
    setUrl: () => {},
})