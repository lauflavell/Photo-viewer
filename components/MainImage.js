import React from "react";

export function MainImage({source}) {
    return(
    <div className="large-img">
        <img src={source} alt="Main"/>
    </div>
    )};