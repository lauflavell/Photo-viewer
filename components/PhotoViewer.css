body {
    margin-left:15vw;
    margin-right: 15vw;
}
/* HEADINGS */

/* If the screen size is 601px wide or more, set the font-size of <div> to 80px */
@media screen and (min-width: 601px) {
    .heading {
        text-align: left;
        font-size: 1.5rem;
    }
    h2 {
        font-size: 1rem;
    
    }
}
  
  /* If the screen size is 600px wide or less, set the font-size of <div> to 30px */
  @media screen and (max-width: 600px) {
    .heading {
        text-align: left;
        font-size: 0.625rem;
    }
    h2 {
        font-size: 0.375rem;
    
    }
}

/* LARGE IMAGE*/

.large-img {
    display: flex;
    list-style: none;
    

}

.large-img img {
    max-width: 100%;
    height: auto;
}

/* THUMBNAIL IMAGES */

.container {
display: flex;
flex-wrap: wrap;
justify-content: space-between;
padding: 0;
margin: 0;
list-style: none;
object-fit: fill;


}

.container .item img{
    max-width:150px;
    width: 100%;
    height: auto;
    
}
.image-border{
    border-color: black;
    border: 4px solid;
    box-sizing: border-box;
    
    
}
.image-no-border {
    border: none;
}

