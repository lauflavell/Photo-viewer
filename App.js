import './App.css';
import { PhotoViewer } from './components/PhotoViewer';
import { ImageUrls } from './components/ImageSelector';
import { useState } from "react";
import React from 'react';
import { MainImage } from './components/MainImage';


function App() {
  const [imgUrl, setUrl] = useState("https://picsum.photos/id/600/1600/900.jpg");

  return (
   
      <main>
          <h1 className="heading">React Photo Viewer</h1>

          <MainImage source={imgUrl}/>

          <h2>Select your photo</h2>

          <div className="container">
            {ImageUrls.map((url,index )=> 
            <PhotoViewer 
            indexKey={index}  
            source= {url} 
            setUrl= {() => setUrl(url)}
            currentImgUrl={imgUrl} />
            )}
          </div>

        </main>
      
  );
}

export default App;
