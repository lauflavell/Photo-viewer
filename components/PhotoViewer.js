
import React from "react";  // import React (to provide access to JSX)
import './PhotoViewer.css';


export function PhotoViewer(props) {    // declare and export new function called 'PhotoViewer'
   const isImageSelected = props.currentImgUrl === props.source;

    let className;
    if (isImageSelected) {
        className = "image-border"
    } else {
        className = "image-no-border"
        }
        return (  
            <li className="item">             
             <img 
             key={props.indexKey} 
             className={className} 
             src={props.source} 
             alt= "Thumbnail" 
             onClick={props.setUrl}/> 
            </li>             
        );
}




