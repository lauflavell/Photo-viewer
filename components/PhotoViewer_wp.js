
import React, { useContext } from "react";  // import React (to provide access to JSX)
import './PhotoViewer.css';
import { ImageUrls } from './components/ImageSelector';
import { ImageContext } from "./ImageContext";

export function PhotoViewer(props) {    // declare and export new function called 'PhotoViewer'
    let { imgUrl, setUrl } = useContext(ImageContext);

    return (  
            <li className="item">
                {ImageUrls.map(url =>              
             <img src={url} alt= "" onClick={() => setUrl(url)} />
             )}
            </li>             
    );
}

