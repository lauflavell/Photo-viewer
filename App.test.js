import { render, screen } from '@testing-library/react';
import App from './App';
import renderer from 'react-test-renderer';
import { PhotoViewer } from './components/PhotoViewer';

/*test('renders learn react link', () => {
  render(<App />);
  const linkElement = screen.getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});*/

describe("My Component", () => {
  it("Should match snapshot without name prop", async () => {
    const tree = renderer.create(<PhotoViewer/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("My Component", () => {
  it("Should match snapshot without name prop", async () => {
    const tree = renderer.create(<App/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
