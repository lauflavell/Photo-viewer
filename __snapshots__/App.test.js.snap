// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Component Should match snapshot without name prop 1`] = `
<li
  className="item"
>
  <img
    alt=""
  />
</li>
`;

exports[`My Component Should match snapshot without name prop 2`] = `
<div>
  <h1
    className="heading"
  >
    React Photo Viewer
  </h1>
  <div
    className="large-img"
  >
    <img
      alt=""
      src="https://picsum.photos/id/600/1600/900.jpg"
    />
  </div>
  <h2>
    Select your photo
  </h2>
  <div
    className="container"
  >
    <li
      className="item"
    >
      <img
        alt=""
        className="border"
        onClick={[Function]}
        src="https://picsum.photos/id/600/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/602/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/603/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/604/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/605/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/606/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/607/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/608/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/609/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/610/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/611/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/612/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/613/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/614/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/615/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/616/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/617/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/618/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/619/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/620/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/621/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/622/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/623/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/625/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/626/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/627/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/628/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/629/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/630/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/631/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/633/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/634/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/635/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/637/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/638/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/639/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/640/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/641/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/642/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/643/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/645/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/646/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/648/1600/900.jpg"
      />
    </li>
    <li
      className="item"
    >
      <img
        alt=""
        onClick={[Function]}
        src="https://picsum.photos/id/649/1600/900.jpg"
      />
    </li>
  </div>
</div>
`;
